### ISSUE TITLE
Add summary information to the validation results

### ISSUE THREAD

**Issue by chrisgorgo** [bids-standard/legacy-validator#137]
As discussed with @constellates (but also requested by @pbellec) the results of the validator (command line and website) should include summary information about the dataset:
- number of subjects
- number of sessions
- available modalities (for example T1w, bold etc.)
- (if _bold) available tasks

**Comment by constellates** [comment]
@chrisfilo I've started working on this and have a couple questions. 

**1:** To capture sessions, subjects, etc I'm iterating the full file list and creating a unique list of all the values for the `ses-`, `sub`, `run-` and `task-` path identifiers. Is this the right strategy, capturing the full set of present values for each identifier even though it's possible all sessions might not match perfectly with regards to the number subjects, runs, tasks, etc?

**2:** It's not mentioned in the original issue but should we include `runs` in the summary?

**3:** To capture the modalities I'm grabbing the suffix on all scans (files ending in `.nii` or `.nii.gz`). Do all scans have this suffix and does the suffix always represent a modality?

You can see the current code on this branch https://github.com/INCF/bids-validator/tree/summary-data and below is an example of the data I'm currently capturing.

```
{
    "sessions": [
        "1",
        "2"
    ],
    "subjects": [
        "01",
        "02",
        "03",
        "04",
        "05",
        "06",
        "07",
        "08",
        "09",
        "10",
        "11",
        "12",
        "13",
        "14",
        "15",
        "16",
        "17",
        "18",
        "19",
        "20",
        "21",
        "22"
    ],
    "runs": [
        "1",
        "2"
    ],
    "tasks": [
        "rest"
    ],
    "modalities": [
        "T1map",
        "T1w",
        "magnitude1",
        "magnitude2",
        "phasediff",
        "bold"
    ]
}
```

**Comment by constellates** [comment]
One more question. Does it make sense to default session and run counts to 1 when no session or run identifiers are found? So a dataset without specified different runs would be said to have 1 run? Or should they be zero or not present at all?

Here's how the final outputs are looking on the web validator and CLI. Let me know if anything looks off to you.

<img width="1439" alt="screen shot 2016-04-27 at 2 19 02 pm" src="https://cloud.githubusercontent.com/assets/4325905/14868660/87d74354-0c83-11e6-85d2-a83f26e4bd5b.png">

<img width="741" alt="screen shot 2016-04-27 at 2 19 22 pm" src="https://cloud.githubusercontent.com/assets/4325905/14868720/e32608b2-0c83-11e6-8823-5e5a02b069bd.png">

If this looks good and I'm handling the above questions correctly I'll add a PR and then push the web validator updates after it's merged in.

**Comment by chrisgorgo** [comment]
This looks great. Some comments:
- Runs are repetitions of the same scan. Counting them is only meaningful in the context of a particular modality (and/or task). We can either skip them altogether or list them next to task and modality labels (with the exception of "bold" which should just list number of tasks if anything).
- Could we use task names from json files instead of task-<name>? In many cases they are more human readable.
- Using suffixes to grab modalities is a great first approach. In the future we can add more parsing to this (for example instead of listing phase1, phase2, magnitute1, magnitude2 we could just say "fieldmap").
- We also need to make sure that the summary generation is robust to datasets that are not valid. I would prefer to display it whenever it's possible (even if there are errors), but at the same time the whole validator should not break because the dataset is invalid and we were trying to generate a summary.

**Comment by constellates** [comment]
I've added a PR and to address the above issues.
- I've removed run tracking for the time being. The way I was capturing it doesn't make sense unless all tasks/modalities always have the same number of runs within a given dataset.
- I switched to getting task names from the JSON files. This is based on my understanding that JSON files with a `TaskName` property are required for task scans and they will always have 'task' in the filename. Let me know if this is incorrect and I'll update the check.
- I'm still using suffixes to get modalities, but have also added a function for grouping modalities. It has groupings specified like this.

```
var modalityGroups = {
    fieldmap: [
        'magnitude1',
        'magnitude2',
        'phase1',
        'phase2'
    ]
};
```

If a dataset has all of the modalities of the group, they will be removed from the list and replaced with the group key name. The fieldmap you mentioned is the only one currently in there. Let me know if you're aware of others I should add. I couldn't find a dataset with the above fieldmap grouping, but 7t_trt has `magnitude1`, `magnitude2` and `phasediff`. Does that also constitute a fieldmap?
- Summary generation should be as robust as the validator as a whole. It will still display when errors and warnings are found.
